### PR TITLE
Sub: Add rangefinder in SITL

### PR DESCRIPTION
**Pull request by patrickelectric**
Also add a non planar sea floor

### CODE REVIEW

File: libraries/SITL/SIM_Submarine.cpp [hunk 1]
```diff
@@ -85,6 +87,19 @@ void Submarine::calculate_forces(const struct sitl_input &input, Vector3f &rot_a
...
+/**
+ * @brief Calculate sea floor depth from submarine position
+ *          This creates a non planar floor for rangefinder sensor test
+ *
+ * @param position
+ * @return float
+ */
+float Submarine::calculate_sea_floor_depth(const Vector3f &position)
+{
+    constexpr float wave_period = 10;
```
Comment (peterbarker) [inline libraries/SITL/SIM_Submarine.cpp]: We have parameters for wave simulation now :-)

Comment (patrickelectric) [reply libraries/SITL/SIM_Submarine.cpp]: Wow, I'll take a look!
Right now I'm removing the wave or the content of this function to allow future improvements in our terrain procedural generation. 
The idea is to provide a good test environment for rangefinder sensors.

**Comment by peterbarker** [comment]
On Fri, 1 Nov 2019, Patrick José Pereira wrote:

> Right now I'm removing the wave or the content of this function to allow future improvements in our terrain procedural generation.
> The idea is to provide a good test environment for rangefinder sensors.

Very interested.  I'm working on a driver for a scanning Sonar - I'd like 
to create a simulated point cloud from the simulated Sonar so we can do 
proximity avoidance based on it.

File: libraries/SITL/SIM_Submarine.cpp [hunk 2]
```diff
@@ -65,8 +65,10 @@ void Submarine::calculate_forces(const struct sitl_input &input, Vector3f &rot_a
         rot_accel += t.rotational * output;
     }
 
+    float floor_depth = calculate_sea_floor_depth(position);
+    range = floor_depth - position.z;
     // Limit movement at the sea floor
-    if (position.z > 100 && body_accel.z > -GRAVITY_MSS) {
+    if (position.z > floor_depth && body_accel.z > -GRAVITY_MSS) {
```
Comment (jaxxzer) [inline libraries/SITL/SIM_Submarine.cpp]: should this be `body_accel.z < -GRAVITY_MSS`?

Comment (jaxxzer) [reply libraries/SITL/SIM_Submarine.cpp]: i see now this logical is not changed in the patch.

**Comment by patrickelectric** [comment]
@jaxxzer conflict fixed.

**Pull request merged**